src/disco/bank/fd_bank_abi.c

```diff
@@ -0,0 +1,418 @@
+  struct __attribute__((__packed__)) __attribute__((aligned(8UL))) {
+    ulong tag; /* Niche tag encoding.  Value 0 is legacy, nonzero is v0. */
+    sanitized_txn_abi_legacy_message1_t legacy;
+  };
+
+  sanitized_txn_abi_v0_loaded_msg_t v0; /* No tag. First field is always non-NULL, so rustc can disciminate from legacy. */
+} sanitized_txn_abi_message_t;
+
+FD_STATIC_ASSERT( sizeof(sanitized_txn_abi_message_t) == 184UL, "messed up size" );
+FD_STATIC_ASSERT( alignof(sanitized_txn_abi_message_t) == 8UL, "messed up size" );
+
+FD_STATIC_ASSERT( offsetof(sanitized_txn_abi_message_t, v0) == 0UL, "messed up size" );
+FD_STATIC_ASSERT( offsetof(sanitized_txn_abi_message_t, legacy) == 8UL, "messed up size" );
+
+struct __attribute__((__packed__)) __attribute__((aligned(8UL))) fd_bank_abi_txn_private {
+  sanitized_txn_abi_signature_t * signatures;
+  ulong signatures_cap;
+  ulong signatures_cnt;
+
+  sanitized_txn_abi_message_t message;
+
+  uchar message_hash[ 32 ];
+  uchar is_simple_vote_tx;
+};
+
+FD_STATIC_ASSERT( sizeof(struct fd_bank_abi_txn_private) == 248UL, "messed up size" );
+FD_STATIC_ASSERT( alignof(struct fd_bank_abi_txn_private) == 8UL, "messed up size" );
+
+FD_STATIC_ASSERT( offsetof(struct fd_bank_abi_txn_private, signatures) == 0UL, "messed up size" );
+FD_STATIC_ASSERT( offsetof(struct fd_bank_abi_txn_private, message) == 24UL, "messed up size" );
+FD_STATIC_ASSERT( offsetof(struct fd_bank_abi_txn_private, message_hash) == 208UL, "messed up size" );
+FD_STATIC_ASSERT( offsetof(struct fd_bank_abi_txn_private, is_simple_vote_tx) == 240UL, "messed up size" );
+
+extern int
+fd_ext_bank_sanitized_txn_load_addresess( void *  bank,
+                                          void *  address_table_lookups,
+                                          ulong   address_table_lookups_cnt,
+                                          void *  out_sidecar );
+
+static uchar BUILTIN_SYSVARS_AND_PROGRAMS[22][32] = {
+  /* Sysvars */
+  { SYSVAR_CLOCK_ID          },
+  { SYSVAR_EPOCH_SCHED_ID    },
+  { SYSVAR_FEES_ID           },
+  { SYSVAR_RECENT_BLKHASH_ID },
+  { SYSVAR_RENT_ID           },
+  { SYSVAR_REWARDS_ID        },
+  { SYSVAR_SLOT_HASHES_ID    },
+  { SYSVAR_SLOT_HIST_ID      },
+  { SYSVAR_STAKE_HIST_ID     },
+  { SYSVAR_INSTRUCTIONS_ID   },
+  { SYSVAR_EPOCH_REWARDS_ID  },
+  { SYSVAR_LAST_RESTART_ID   },
+
+  /* Programs */
+  { CONFIG_PROG_ID          },
+  { FEATURE_ID              },
+  { NATIVE_LOADER_ID        },
+  { STAKE_PROG_ID           },
+  { STAKE_CONFIG_PROG_ID    },
+  { VOTE_PROG_ID            },
+  { SYS_PROG_ID             },
+  { BPF_LOADER_1_PROG_ID    },
+  { BPF_LOADER_2_PROG_ID    },
+  { BPF_UPGRADEABLE_PROG_ID },
+};
+
+static int
+is_builtin_key_or_sysvar( uchar * key ) {
+  for( ulong i=0; i<sizeof(BUILTIN_SYSVARS_AND_PROGRAMS)/sizeof(*BUILTIN_SYSVARS_AND_PROGRAMS); i++ ) {
+    if( !memcmp( key, BUILTIN_SYSVARS_AND_PROGRAMS[i], 32UL ) ) return 1;
+  }
+
+  return 0;
+}
```

Comment:
You are right that we still need this here to check the accounts in the address lookup table. The list in `fd_pack.c` is almost correct. I'll work on fixing it